daffodil-runtime1/src/main/scala/edu/illinois/ncsa/daffodil/processors/Evaluatable.scala

```diff
@@ -336,30 +292,40 @@ abstract class Evaluatable[+T <: AnyRef](protected val rd: RuntimeData, qNameArg
     }
   }
 
-  private var constValue_ : Option[AnyRef] = None
-  protected def constValue = constValue_.asInstanceOf[Option[T]]
+  private var constValue_ : Maybe[AnyRef] = Nope
 
-  def optConstant = {
-    ensureCompiled
-    constValue
-  }
+  /**
+   * Preferred for use in the runtime.
+   */
+  @inline final def maybeConstant = constValue_.asInstanceOf[Maybe[T]]
```

Comment:
Why the switch from Option to Maybe?

Reply:
Seemed inconsistent with general notion that Option is used in compiler and Maybe in the runtime. Saves 1 indirection.